evap/rewards/views.py

```diff
@@ -65,10 +65,13 @@ def semester_reward_points(request, semester_id):
 
     data = []
     for participant in participants:
-        number_of_courses = Course.objects.filter(semester=semester, participants=participant).count()
-        number_of_courses_voted_for = Course.objects.filter(semester=semester, voters=participant).count()
+        number_of_required_courses = Course.objects.filter(semester=semester, participants=participant, is_required_for_reward=True).count()
+        number_of_required_courses_voted_for = Course.objects.filter(semester=semester, voters=participant, is_required_for_reward=True).count()
+        number_of_optional_courses = Course.objects.filter(semester=semester, participants=participant, is_required_for_reward=False).count()
+        number_of_optional_courses_voted_for = Course.objects.filter(semester=semester, voters=participant, is_required_for_reward=False).count()
         earned_reward_points = RewardPointGranting.objects.filter(semester=semester, user_profile=participant).exists()
-        data.append((participant, number_of_courses_voted_for, number_of_courses, earned_reward_points))
+        data.append((participant, number_of_required_courses_voted_for, number_of_required_courses,
+            number_of_optional_courses_voted_for, number_of_optional_courses, earned_reward_points))
```

Comment:
this smells like (#num participants in semester) * 5 queries, i'd expect this page to be quite slow on production. how frequently is it used? do you think the performance is acceptable?